File: 3oMod/Desafios/desafio1/desafio.js
// Desafio
// Você precisa criar um programa que calcule a soma de todos os números entre 1 e um número fornecido pelo usuário, exceto aqueles que são divisíveis por 3.
// Entrada
// O programa deve receber a entrada do usuário, que deve ser um número inteiro positivo. Em seguida, o programa deve calcular a soma de todos os números entre 1 e o número fornecido pelo usuário, exceto aqueles que são divisíveis por 3.
// Saída
// A saída do programa deve imprimir um texto com a soma dos números de 1 até o número fornecido pelo usuário (exceto aqueles divisíveis por 3).





//Desafios JavaScript na DIO têm funções "gets" e "print" acessíveis globalmente:
//- "gets" : lê UMA linha com dado(s) de entrada (inputs) do usuário;
//- "print": imprime um texto de saída (output), pulando linha.

let numero = parseInt(gets());

let soma = 0;
for (let i = 1; i <= numero; i++) {
  if (i % 3 !== 0) {
    soma += i;
  }
}

console.log(`Soma dos números de 1 a ${numero}, exceto os divisíveis por 3: ${soma}`);

File: 3oMod/Desafios/desafio3/desafio.cs
// Desafio
// Um número é azarado se contém um número 1 seguido por um 3 entre seus dígitos, ou seja, sempre que houver o caracteres "13", ele será considerado um número azarado. Por exemplo, o número 341329 é de má sorte, enquanto o número 12321 não é.
// Dado um inteiro N, seu programa terá que determinar se N é azarado ou não.
// Entrada
// A entrada consiste em um número positivo N (0 <= N <= 10¹⁷).
// Saída
// Imprima a mensagem "N es de Mala Suerte" se N é de má sorte, caso contrário imprima "N NO es de Mala Suerte".


// Para ler e escrever dados em C#, utilizamos os seguintes métodos da classe Console:
// - Console.ReadLine: lê UMA linha com dado(s) de Entrada (Inputs) do usuário;
// - Console.WriteLine: imprime um texto de Saída (Output) e pulando uma linha.

using System;

public class Program 
{
    public static void Main() 
    {
        string numero = Console.ReadLine();

        bool azarado = false;

        // Verifica se o número contém a sequência "13"
        for (int i = 0; i < numero.Length - 1; i++) 
        {
            if (numero[i] == '1' && numero[i + 1] == '3') 
            {
                azarado = true;
                break;
            }
        }

        // Imprime o resultado de acordo com a condição
        if (azarado)
            Console.WriteLine($"{numero} es de Mala Suerte");
        else
            Console.WriteLine($"{numero} NO es de Mala Suerte");
    }
}


